lib/liquid/tags/include.rb

```diff
@@ -49,6 +66,7 @@ def parse(_tokens)
     def render_to_output_buffer(context, output)
       template_name = context.evaluate(@template_name_expr)
       raise ArgumentError, options[:locale].t("errors.argument.include") unless template_name
+      raise ArgumentError unless allowed_path?(template_name)
```

Comment:
I don't think this should be the responsibility of the tag, instead, it should be the responsibility of the filesystem to reject access. Another way to accomplish this could be to pass through the caller in this case `include` through to `PartialCache.load` likely though either `context` or `parse_context` and all this logic can live inside the FileSystem and implement its own authentication.

Reply:
For example, the base user could be  
```
context.registers[`current_user`] = 'system'
```
and then when you enter a scope that you want to restrict
```
context.registers[`current_user`] = 'restricted_user'
```
This is all obviously a very basic example, but then inside the `File System` you can do a check if the current user/callee has access to the files you are trying to access, if not throw an exception up.